### PR TITLE
[move-examples] Remove legacy entry fun from marketplace example

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
This cleans up the usage of `legacy_entry_fun` in the example

### Test Plan
https://nft-playground.gnazar.io/ is able to use it correctly to handle listings and purchases

### CODE REVIEW

**Review by movekevin — COMMENTED** [review]
#[legacy_entry_fun] is used to allow these entry funcs to return values and be used on chain directly without having to write redundant non-entry public functions. I'd keep these for now until we're done experimenting with that

**Review by bowenyang007 — APPROVED** [review]

File: aptos-move/move-examples/marketplace/sources/collection_offer.move [hunk 1]
```diff
@@ -92,14 +91,33 @@ module collection_offer {
...
+            creator_address,
+            collection_name,
+            fee_schedule,
+            item_price,
+            amount,
+            expiration_time
+        );
+    }
+
+    public fun init_for_tokenv1_internal<CoinType>(
```
Comment (movekevin) [inline aptos-move/move-examples/marketplace/sources/collection_offer.move:106]: _internal is a misleading name for a public function. I prefer "_entry" for the entry function and no suffix for the non-entry public function

**Review by movekevin — APPROVED** [review]
Approved except for one minor naming comment

**Comment by davidiw** [comment]
landing this as  part of a stack.

**Pull request closed without merging**